src/components/filterPatterns/modal/OpenFilterButton.tsx

```diff
@@ -16,31 +16,180 @@ import {
 
 import { FilterPatternProps } from '../props';
 
-export type PaddingX = '0' | 'md';
-export type ResetButtonVariant = 'circle' | 'square';
+export type OpenFilterButtonPaddingX = '0' | 'md';
+export type OpenFilterButtonDisplayType = 'default' | 'inline';
 type Variant = 'sm' | 'md';
 type Props = Pick<
   FilterPatternProps,
   'label' | 'displayValue' | 'Icon' | 'isApplied'
 > &
   Pick<ButtonProps, 'backgroundColor' | 'color'> & {
+    appliedLabel?: string;
     onClick: () => void;
     variant?: Variant;
     isDisabled?: boolean;
     height?: ResponsiveValue<string>;
-    paddingX?: PaddingX;
+    paddingX?: OpenFilterButtonPaddingX;
     showResetButton?: boolean;
-    resetButtonVariant?: ResetButtonVariant;
     resetButtonAriaLabel?: string;
     onResetFilter?: () => void;
+    displayType?: OpenFilterButtonDisplayType;
   };
 
 const heightFromVariant: Record<Variant, ResponsiveValue<string>> = {
   sm: 'md',
   md: 'lg',
 };
 
+const getIconColor = (
+  isDisabled: boolean,
+  color?: ResponsiveValue<string>,
+): string => {
+  if (isDisabled) return 'gray.300';
+  const colorValue = typeof color === 'string' ? color : undefined;
+  return colorValue || 'gray.500';
+};
+
+const getButtonColor = (
+  isDisabled: boolean,
+  color?: ResponsiveValue<string>,
+): ResponsiveValue<string> => {
+  if (color) return color;
+  return isDisabled ? 'gray.300' : 'gray.900';
+};
+
+const getResetButtonConfig = (
+  isInline: boolean,
```

Comment:
could we go with `displayType?: OpenFilterButtonDisplayType;` instead of boolean flag?